gocourse02/main.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"fmt"
+	"math/rand/v2"
+)
+
+/*
+Написати програму “Зоопарк”. Звіри повтікали (більше трьох штук), наглядач повинен їх зібрати.
+Кожна сутність (наглядач, звір, клітка, …) представляється окремою структурою (zookeeper, animal, cage, …).
+Треба використати щонайменше: структури, вказівник, nil, будування, конструктор. Додати тваринам можливість
+розмножуватись. Програма має демонструвати свою роботу через вивід в stdout.
+*/
+
+const escapedAnimalsCount = 15 // Кількість звірів які повтікали
+//type Gender string
+//
+//const (
+//	male   Gender = "male"
+//	female Gender = "female"
+//)
+
+type ZooKeeper struct {
+	Name          string
+	AnimalsFound  int
+	AnimalsCaught int
+}
+
+func (k *ZooKeeper) SearchingForAnimalsReturnToCage(catch Catcher, animal *Animal, cage *Cage) {
+	if catch.FindEscapedAnimal() {
+		k.AnimalsFound++
+		if catch.AttemptCatchAnimal(cage, animal) {
+			k.AnimalsCaught++
+			cage.CurrentSeats++
+		}
+	}
+}
+
+type Catcher interface {
+	FindEscapedAnimal() bool                // Спроба знайти звіра
+	AttemptCatchAnimal(*Cage, *Animal) bool // Сброба зловити звіра в клітку
+}
+
+type Animal struct {
+	ID     int
+	Name   string
+	Gender string
+	Weight int
+}
+
+type Cage struct {
+	MaxSeats      int
+	MaxWeight     int
+	CurrentSeats  int
+	CurrentWeight int
+	Animal        []Animal
+}
+
+// Шукаємо звіра. Результат = Знаходимо або ні
+func (k *ZooKeeper) FindEscapedAnimal() bool {
+	return rand.N(2) == 1
+}
+
+// Намагаемося зловити в клітку знайденого звіра, якщо в клітці є місця  і не перевищує максимально дозволену вагу
+func (k *ZooKeeper) AttemptCatchAnimal(cage *Cage, a *Animal) bool {
```

Comment:
Просто для консистентності параметрів:
```suggestion
func (k *ZooKeeper) AttemptCatchAnimal(cage *Cage, animal *Animal) bool {
```

Бо в нас клітка "cage", то й звір вже має бути повністю "animal".